File: microservers/immobilierservice/src/main/java/io/multi/immobilierservice/domain/TypeBien.java
package io.multi.immobilierservice.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TypeBien {

    private Long typeBienId;
    private String typeBienUuid;
    private String code;               // MAISON | APPARTEMENT | IMMEUBLE | TERRAIN | BUREAU | BOUTIQUE | CHAMBRE
    private String libelle;
    private String description;
    private String icone;
    private Integer ordreAffichage;
    private Boolean actif;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}

File: mobile_app/lib/features/immo/models/type_bien.dart
/// Type de bien immobilier (référentiel). Backend : `GET /immo/types-bien`
/// retourne `{data: {types: [{...}]}}`. Le `code` (MAISON, APPARTEMENT, ...)
/// est l'identifiant logique stable utilisé pour filtrer côté search.
/// `icone` est un nom Material Icons (ex: "home", "apartment").
///
/// Champs backend ignorés en 15.2b : typeBienUuid, ordreAffichage, actif,
/// createdAt, updatedAt.
class TypeBien {
  final int typeBienId;
  final String code;
  final String libelle;
  final String? description;
  final String? icone;

  const TypeBien({
    required this.typeBienId,
    required this.code,
    required this.libelle,
    this.description,
    this.icone,
  });

  factory TypeBien.fromJson(Map<String, dynamic> json) => TypeBien(
        typeBienId: json['typeBienId'] as int,
        code: json['code'] as String,
        libelle: json['libelle'] as String,
        description: json['description'] as String?,
        icone: json['icone'] as String?,
      );
}
